### PR TITLE
[types] Deprecate Transaction::Program

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

This diff is going to use our new model of Transaction::Script and Transaction::Module to do simple payment transactions in the libra codebase. Previously the normal transaction has one type of payload: Transaction::Program and one can publish a module AND execute some script within one single transaction. This feature however, is complicating the VM a lot and can really buy us few things as the modules are stateless. In our new model, we will have two types of transactions: Transaction::Module which one can use to publish a module and do nothing else, and Transaction::Script which one can execute code that mutates the resources on chain.

This diff will NOT be a breaking change as we still keep the functionality of processing Transaction::Program in the VM so that we will have some time to break the backward incompatibility there.

I removed the repl in this PR as well. Reason being that the local cli will have a superset of the functionality provided by the repl. 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: language/compiler/src/lib.rs [hunk 1]
```diff
@@ -89,18 +89,12 @@ impl<'a> Compiler<'a> {
     }
 
     /// Compiles the code and arguments into a `Program` -- the bytecode is serialized.
-    pub fn into_program(mut self, args: Vec<TransactionArgument>) -> Result<Program> {
+    pub fn into_program(mut self, args: Vec<TransactionArgument>) -> Result<Script> {
```
Comment (dariorussi) [inline language/compiler/src/lib.rs]: should we call this into_script? and fix the comment as well?

File: testsuite/tests/libratest/smoke_test.rs [hunk 2]
```diff
@@ -393,25 +393,22 @@ fn test_external_transaction_signer() {
...
+                    assert_eq!(arg_amount.clone(), amount);
+                }
+                _ => panic!(
+                    "The first argument for payment transaction must be recipient address \
+                     and the second argument must be amount."
+                ),
+            },
+            _ => panic!("Signed transaction payload arguments must have two arguments."),
+        },
         _ => panic!("Signed transaction payload expected to be of struct Program"),
```
Comment (dariorussi) [inline testsuite/tests/libratest/smoke_test.rs]: change Program to Script?

File: types/src/test_helpers/transaction_test_helpers.rs [hunk 3]
```diff
@@ -16,18 +16,45 @@ static PLACEHOLDER_SCRIPT: &[u8] = include_bytes!("fixtures/scripts/placeholder_
...
+}
+
 // Test helper for transaction creation
 pub fn get_test_signed_transaction(
     sender: AccountAddress,
     sequence_number: u64,
     private_key: Ed25519PrivateKey,
     public_key: Ed25519PublicKey,
-    program: Option<Program>,
+    program: Option<Script>,
```
Comment (dariorussi) [inline types/src/test_helpers/transaction_test_helpers.rs]: should we also change all those variables from program to script?

File: language/compiler/src/lib.rs (hunk 1 above)
Comment (runtian-zhou) [reply language/compiler/src/lib.rs]: The reason why I didn't rename it to into_script is because we've already have `input_script` in the scope.

**Comment by phoenix-antigravity** [comment]
CLI part lgtm! I see that syntax has changed a bit. Should we update our online documentation? cc: @archjayas

**Comment by davidiw** [comment]
It might be worth updating this file to: https://github.com/libra/libra/tree/master/common/canonical_serialization

**Pull request merged**